lib/common/common/src/counter/conditioned_counter.rs

```diff
@@ -0,0 +1,114 @@
+use std::ops::Deref;
+
+use super::hardware_counter::HardwareCounterCell;
+use super::hardware_data::HardwareData;
+
+/// A counter that measures or disposes measurements based on a condition.
+/// This is needed in places where we need to decide at runtime whether to measure or not.
+/// Implements `Deref<Target=HardwareCounterCell>` so it can be directly passed to functions
+/// that need a `HardwareCounterCell` as parameter.
+pub struct ConditionedCounter<'a> {
```

Comment:
This does not seem to be used yet, or is it?

Reply:
Nevermind, I see it is used in a sub-PR: https://github.com/qdrant/qdrant/pull/6296